sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -662,15 +672,20 @@ sycl::detail::pi::PiProgram ProgramManager::getBuiltPIProgram(
     // Save program to persistent cache if it is not there
     if (!DeviceCodeWasInCache)
       PersistentDeviceCodeCache::putItemToDisc(
-          Device, Img, SpecConsts, CompileOpts + LinkOpts, BuiltProgram.get());
+          Device, Img, SpecConsts,
+          std::string_view(CompileOptsString + LinkOptsString),
+          BuiltProgram.get());
+
     return BuiltProgram.release();
   };
 
   uint32_t ImgId = Img.getImageID();
+  std::string_view CompileOptsUpdated =
+      CompileOptsString.empty() ? CompileOpts
+                                : std::string_view(CompileOptsString);
```

Comment:
Why is this distinction needed here? Aren't they the same thing at this point in the code?

Reply:
No, CompileOpts is what we get from applyCompileOptionsFromEnvironment() it does not point to the string that was updated inside BuildF. If it was changed inside BuildF we would want it to point to the updated value.

Reply:
Thanks for catching that. Restored to the original behavior.